views/event/index.ejs

```diff
@@ -7,59 +7,122 @@
-                    all_user_role.ProjectManager,
-                    all_user_role.Admin,
-                    all_user_role.EventExecutive,
-                    all_user_role.ProjectLeader], user_role)) { %>
-                    <div class="col-sm-8-offset" align="right">
-                        <a class="btn primary-button" href="/event/create">Create Event</a>
+    <% if (eventsSet.length == 0){ %>
+        <p class="error-msg">Sorry! Currently, there are no events available</p>
+    <% } else { %>
+        <div class="cover-flow">
+            <% if (eventsSet.length >= 3) { %>
+                <div class="top-stack">
+                    <% for (var i = 0; i <= Math.floor(eventsSet.length / 2) - 1; i++) { %>
+                        <img class="top-img sl-img" onclick="clickUp(<%= i %>)" id="event<%= i %>" src="<%= eventsSet[i].image %>" alt="<%=
+                            eventsSet[i]
+                                    .description
+                        %>"
+                             onerror="this.src='/img/icon.jpg'">
+                    <% } %>
+                </div>
+                <div class="center-stack">
+                    <img class="curr-img sl-img" id="event<%= Math.floor(eventsSet.length / 2) %>"
+                         src="<%= eventsSet[Math.floor(eventsSet.length /
+                                 2)]
+                                 .image %>"
+                         alt="<%= eventsSet[Math.floor(eventsSet.length / 2)].description %>"
+                         onerror="this.src='/img/icon.jpg'">
+                </div>
+                <div class="bot-stack">
+                    <% for (var i = eventsSet.length - 1; i > Math.floor(eventsSet.length / 2); i--) { %>
+                        <img class="bot-img sl-img" onclick="clickDown(<%= i %>)" id="event<%= i %>" src="<%= eventsSet[i].image %>"
+                             alt="<%= eventsSet[i].description %>"
+                             onerror="this.src='/img/icon.jpg'">
+                    <% } %>
+                </div>
+            <% } else { %>
+                <div class="top-stack">
+                </div>
+                <div class="center-stack">
+                    <img class="curr-img sl-img" id="event0" src="<%= eventsSet[0].image %>" alt="<%= eventsSet[0].description %>"
+                         onerror="this.src='/img/icon.jpg'">
+                </div>
+                <div class="bot-stack">
+                    <% if (eventsSet.length === 2) { %>
+                        <img class="bot-img sl-img" id="event1" onclick="clickDown(<%= 1 %>)" src="<%= eventsSet[1].image %>"
+                             alt="<%= eventsSet[1].description %>"
+                             onerror="this.src='/img/icon.jpg'">
+                    <% } %>
+                </div>
+            <% } %>
+        </div>
+        <% if (eventsSet.length >= 3) { %>
+            <% for (var i = 0; i < eventsSet.length; i++) { %>
```

Comment:
Always show the nearest upcoming event, and combine these two sections to prevent redundant code.